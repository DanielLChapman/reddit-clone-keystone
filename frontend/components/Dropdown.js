import React, { useState } from 'react';
import PropTypes from 'prop-types'

function Dropdown(props) {
    /*
        let options = ['Hot', 'Best', 'New', 'Top', 'Rising'];
        const [dropValue, setDropValue] = useState('Hot');
        <Dropdown returnFunc={setDropValue} selected={dropValue} options={options} />    
    */
    return (
        <select value={props.selected} onChange={(e) => {
            props.returnFunc(e.target.value);
        }}>
            {
                props.options && props.options.map((x, i) => (
                    <option key={i} value={x}>{x}</option>
                ))
            }
        </select>
    );
}

Dropdown.propTypes = {
    returnFunc: PropTypes.func.isRequired,
    options: PropTypes.array.isRequired,
    selected: PropTypes.string.isRequired,
};
  

export default Dropdown;