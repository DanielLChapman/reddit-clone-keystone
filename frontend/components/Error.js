import React from 'react';

function Error(props) {
    return (
        <span>
            {props.error}
        </span>
    );
}

export default Error;