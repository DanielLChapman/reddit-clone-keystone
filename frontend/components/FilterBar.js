import React, { Component } from 'react';
import { FaHotjar, FaBahai, FaGopuram, FaChartLine} from 'react-icons/fa';



class FilterBar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            li1: true,
            li2: false,
            li3: false,
            li4: false
        }
    }

    toggleClass = (num) => {
        let state = {
            li1: false,
            li2: false,
            li3: false,
            li4: false
        };
        state['li'+num] = true;
        this.setState({...state});
    }

    render() {
        let style = {
            display: "none"
        };
        if (this.state.li3) {
            style= {
                display: "inline-block"
            };
        }
        return (
            <section className="filter-bar"> 
                <h5>Popular Posts</h5>
                <ul>
                    <li className={this.state.li1 ? 'active' : null} onClick={() => this.toggleClass(1)}><FaHotjar /> Hot</li>
                    <li className={this.state.li2 ? 'active' : null} onClick={() => this.toggleClass(2)}><FaBahai /> New</li>
                    <li className={this.state.li3 ? 'active' : null} onClick={() => this.toggleClass(3)}><FaGopuram /> Top</li>
                    <li className="time-selector-for-filter-bar" style={style}>Today</li>
                    <li className={this.state.li4 ? 'active' : null} onClick={() => this.toggleClass(4)}><FaChartLine /> Rising</li>
                </ul>
            </section>
        );
    }
}

export default FilterBar;