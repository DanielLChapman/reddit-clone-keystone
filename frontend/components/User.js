import { gql, useQuery } from '@apollo/client';

export const CURRENT_USER_QUERY = gql`
  query {
    authenticatedItem {
      ... on User {
        id
        username
        email
        name
        posts {
          votes {
            vflag
          }
        }
        comments {
          votes {
            vflag
          }
        }
        moderating {
          id
          slug
          name
        }
        owner {
          id
          slug
          name
        }
        subreddits {
          id
          slug
          name
        }
        postvotes {
          post {
            title
            id
          }
          id
          vflag
        }
        commentvotes {
          comment {
            id
          }
          id
          vflag
        }
    }
  }
  }
`;

export function useUser() {

  const { data } = useQuery(CURRENT_USER_QUERY);
  let test = {...data};

  
  if (test.authenticatedItem) {
    let x = Object.assign({}, test.authenticatedItem);
    //count up post votes
    let postVotes = 0;
    let commentVotes = 0;
    
    x.posts.forEach((q) => {
      q.votes.forEach((y) => {
        y.vflag === 'Upvote' ? postVotes += 1 : postVotes -= 1;
      })
    });
    //count up comment votes
    x.comments.forEach((q) => {
      q.votes.forEach((y) => {
        y.vflag === 'Upvote' ? commentVotes += 1 : commentVotes -= 1;
      })
    });

    x.postVotes = postVotes;
    x.commentVotes = commentVotes;

    return x;
  } else {
    return data?.authenticatedItem;
  }

  
}
