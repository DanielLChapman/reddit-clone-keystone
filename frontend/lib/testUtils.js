import casual from 'casual';

  
// seed it so we get consistent results
casual.seed(777);

const fakeUser = (overrides) => (
    {
    __typename: 'User',
    id: '4234',
    name: casual.name,
    username: casual.name,
    owner: [],
    email: casual.email,
    permissions: ['ADMIN'],
    moderating: [],
    subreddits: [],
    posts: [],
    postvotes: [],
    firstSignin: false,
    comments: [],
    commentvotes:[],
    ...overrides,
});

const fakeSubreddit = (overrides) => (
    {
        __typename: 'Subreddit',
        id: casual.uuid,
        name: casual.string,
        title: casual.string,
        slug: casual.string,
        sidebar: casual.string,
        rules: casual.string,
        description: casual.string,
        status: 'PUBLIC',
        ownder: fakeUser(),
        moderators: [],
        subscriber: [],
        posts: [],
        createdAt: Date.now(),
    }
);

const fakeLinkPost = (overrides) => (
    {
    __typename: 'Post',
    title: casual.sentence,
    user: fakeUser(),
    subreddit: fakeSubreddit(),
    createdAt: Date.now(),
    votes: [],
    post_slug: casual.string,
    link: 'www.google.com',
    type: 'Link',
    comments: [],
    removed: 'False',
    id: casual.uuid,


});

const fakeMediaPost = (overrides) => (
    {
    __typename: 'Post',
    title: casual.sentence,
    user: fakeUser(),
    subreddit: fakeSubreddit(),
    createdAt: Date.now(),
    votes: [],
    post_slug: casual.string,
    link: 'https://cdn.fileinfo.com/img/ss/sm/png_79.jpg',
    type: 'Link',
    comments: [],
    id: casual.uuid,
    removed: 'False'

});

const fakeTextPost = (overrides) => (
    {
    __typename: 'Post',
    title: casual.sentence,
    content: casual.sentence,
    user: fakeUser(),
    subreddit: fakeSubreddit(),
    createdAt: Date.now(),
    votes: [],
    post_slug: casual.string,
    link: '',
    type: 'Text',
    comments: [],
    id: casual.uuid,
    removed: 'False'

});

const fakeComment = (overrides) => (
    {
    __typename: 'Comment',
    content: casual.sentence,
    user: fakeUser(),
    post: fakeTextPost(),
    createdAt: Date.now(),
    votes: [],
    parent: null,
    id: casual.uuid,
    children: [],

});


// Fake LocalStorage
class LocalStorageMock {
    constructor() {
      this.store = {};
    }
  
    clear() {
      this.store = {};
    }
  
    getItem(key) {
      return this.store[key] || null;
    }
  
    setItem(key, value) {
      this.store[key] = value.toString();
    }
  
    removeItem(key) {
      delete this.store[key];
    }
  }



export {

    LocalStorageMock,
    fakeLinkPost,
    fakeComment,
    fakeMediaPost,
    fakeSubreddit,
    fakeTextPost,
    fakeUser,

  };