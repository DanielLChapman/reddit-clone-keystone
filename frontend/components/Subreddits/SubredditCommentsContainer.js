import React from 'react';
import PropTypes from 'prop-types';
import gql from 'graphql-tag';
import { FaLongArrowAltUp, FaLongArrowAltDown } from 'react-icons/fa';
import { useMutation, useQuery } from '@apollo/client';
import { CREATE_VOTE, DELETE_VOTE, UPDATE_VOTE } from '../SmallPosts/PostMain';
import { CURRENT_USER_QUERY, useUser } from '../User';
import CommentsPage from './SubredditComments/CommentsPage';
import SubredditTopBar from './SubredditComments/CommentsPageTopBar';

const GET_POST_INFO = gql`
    query GET_POST_INFO($id: ID!) {
        Post( where: { id: $id } )
    {
        id
        content
        title
        user {
            id
            name
            username
        }
        createdAt
        votes {
            vflag
        }
        post_slug
        link
        removed
        subreddit {
            slug
        }
        type
        comments {
            id
            parent {
                id
            }
            content
            createdAt
            user {
                id
                username
            }
            votes {
                vflag
            }
        }

    }
    }
`;

const GET_SPECIFIC_COMMENTS = gql`
    query GET_SPECIFIC_COMMENTS($id: ID!) {
        Comment (
            where: {
                id: $id
            }
        ) {
            id
            parent {
                id
            }
            content
            createdAt
            user {
                id
                username
            }
            votes {
                vflag
            }
            children {
                id
                parent {
                    id
                }
                content
                createdAt
                user {
                    id
                    username
                }
                votes {
                    vflag
                }
                children {
                    id
                    parent {
                        id
                    }
                    content
                    createdAt
                    user {
                        id
                        username
                    }
                    votes {
                        vflag
                    }
                    children {
                        id
                        parent {
                            id
                        }
                        content
                        createdAt
                        user {
                            id
                            username
                        }
                        votes {
                            vflag
                        }
                        children {
                            id
                            parent {
                                id
                            }
                            content
                            createdAt
                            user {
                                id
                                username
                            }
                            votes {
                                vflag
                            }
                        }
                    }
                }
            }
        } 
    }
`;

function SubredditCommentsContainer(props) {
    const user = useUser();

    const {data, error, loading} = useQuery(GET_POST_INFO, {
        variables: {
            id: props.postid,
        }
    });

    

    if (error) return <span>Error...</span>
    if (loading) return <span>Loading...</span>

    let post = {...data?.Post};
    let style = {
        display: 'block'
    }
    let showLinkToGoBackToAllComments = false;

    if (props.commentArray) {
        post.comments = props.commentArray;
        showLinkToGoBackToAllComments = true;
        style = {
            display: 'none'
        }
    }

    return (
        <div className="comment-page">
            <SubredditTopBar user={user} post={post} subreddit={props.subreddit}/>
            <section className="comment-container" >
                
                <CommentsPage showLinkToGoBackToAllComments={showLinkToGoBackToAllComments} propstyle={style} ownership={props.ownership} user={user} post={post} subreddit={props.subreddit} />
            </section>
        </div>
    );
}

// SubredditCommentsContainer.propTypes = {
//     subreddit: PropTypes.object.isRequired,
//     postid: PropTypes.string.isRequired,

// }

export default SubredditCommentsContainer;