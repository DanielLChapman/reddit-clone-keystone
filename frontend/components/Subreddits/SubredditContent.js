import Link from 'next/link';
import React from 'react';

import PropTypes from 'prop-types';
import SubredditPostForm from './SubredditPostForm';
import SubredditHomePage from './SubredditHomePage';
import SubredditCommentsContainer from './SubredditCommentsContainer';
import SubredditMediaForm from './SubredditMediaForm';
import { faIgloo } from '@fortawesome/free-solid-svg-icons';
import { useUser } from '../User';
import checkUserToSubreddit from '../../lib/checkUser';


function SubredditContent(props) {
    let user = useUser();

    let ownership = checkUserToSubreddit(user, props.subreddit.moderators, props.subreddit.owner);

    switch(props.type) {
        case 'Home': 
            if (props.slug && props.subreddit) {
                return <SubredditHomePage ownership={ownership || false} slug={props.slug} id={props.subreddit.id} subreddit={props.subreddit}/>
            }
            return <span>Something was missing here</span>
        case 'comment':
            if (props.slug && props.subreddit && props.postslug && props.postid) {
                return <SubredditCommentsContainer ownership={ownership || false} slug={props.slug} subreddit={props.subreddit} postslug={props.postslug} postid={props.postid}/>
            }
            return <span>Something was missing here</span>
        case 'permalinkedcomments':
            if (props.slug && props.subreddit && props.postslug && props.postid && props.commentid && props.commentArray) {
                return <SubredditCommentsContainer commentid={props.commentid} commentArray={props.commentArray} ownership={ownership || false} slug={props.slug} subreddit={props.subreddit} postslug={props.postslug} postid={props.postid}/>
            }
            return <span>Something was missing here!</span>
        case 'form':
            if (props.selftext && props.selftext === "true") {
                return (
                    <SubredditPostForm slug={props.slug} id={props.subreddit.id} />
                )
            }
            return (
                <SubredditMediaForm slug={props.slug} id={props.subreddit.id} />
            )
        default: 
            return (
                <span>Something went wrong...</span>
            )
    }
    

}


export default SubredditContent;