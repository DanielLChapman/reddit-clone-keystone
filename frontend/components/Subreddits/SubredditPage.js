import { useQuery } from '@apollo/client';
import gql from 'graphql-tag';
import React from 'react';
import SubredditContent from './SubredditContent';

const GET_SUBREDDIT_INFO = gql`
    query GET_SUBREDDIT_INFO($slug: String!) {
        allSubreddits(where: {
            slug: $slug
        }) {
            name
            title
            sidebar
            description
            status
            owner {
                name
            }
            moderators {
                name
            }
        }
    }
`;

function SubredditPage(props) {

    const {data, error, loading} = useQuery(GET_SUBREDDIT_INFO, {
        variables: {
            slug: props.slug
        }
    });
    if (loading) return <span>Loading...</span>
    if(error) return <span>Nothing To See Here</span>

    console.log(data);
    return (
        <div className="subreddit-content">
            <div className="subreddit-left">
                {/* HArd coding news id, will grab from gql later */}
                <SubredditContent slug={props.slug} type={props.type} selftext={props.selftext} id='60944706438ea508cdf752c9' /> 

            </div>
            <div className="subreddit-right">
                blah
            </div>
        </div>
    );
}

export default SubredditPage;