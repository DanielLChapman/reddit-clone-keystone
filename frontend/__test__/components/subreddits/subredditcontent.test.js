import { render, screen } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import SubredditContent from '../../../components/Subreddits/SubredditContent';
import { CURRENT_USER_QUERY } from '../../../components/User';
import SubredditPage, { GET_DOG_QUERY, GET_SUBREDDIT_INFO } from '../../../components/Subreddits/SubredditPage';
import { fakeSubreddit, fakeTextPost, fakeUser  } from '../../../lib/testUtils';
import wait from 'waait';
import { act } from 'react-dom/test-utils';
import userEvent from '@testing-library/user-event';
import { GET_POST_INFO } from '../../../components/Subreddits/SubredditCommentsContainer';
//import Signup, { SIGNUP_MUTATION } from '../../components/SignUp';

const fakeSubredditObj = fakeSubreddit();
const fakePostObj = fakeTextPost();

const signedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: fakeUser() } },
    },
  ];

const mocks = [
      {request: {
        query: GET_SUBREDDIT_INFO,
        variables: { slug: fakeSubredditObj.slug },
      },
      result: () => {
          return {
        data: { allSubreddits: [{ 

            id: fakeSubredditObj.id,
            name: fakeSubredditObj.name,
            title: fakeSubredditObj.title,
            slug: fakeSubredditObj.slug,
            sidebar: fakeSubredditObj.sidebar,
            description: fakeSubredditObj.description,
            status: 'Public',
            owner: {
                username: fakeSubredditObj.owner.username,
            },
            moderators: [],
            createdAt: fakeSubredditObj.createdAt,
        } ]},
      }},
    },
    {request: {
        query: GET_SUBREDDIT_INFO,
        variables: { slug: fakePostObj.subreddit.slug },
      },
      result: () => {
          return {
        data: { allSubreddit: [{ 

            id: fakePostObj.subreddit.id,
            name: fakePostObj.subreddit.name,
            title: fakePostObj.subreddit.title,
            slug: fakePostObj.subreddit.slug,
            sidebar: fakePostObj.subreddit.sidebar,
            description: fakePostObj.subreddit.description,
            status: 'Public',
            owner: {
                username: fakePostObj.subreddit.owner.username,
            },
            moderators: [],
            createdAt: fakePostObj.subreddit.createdAt,
        }] },
      }},
    },
    {
        request: {
            query: GET_POST_INFO,
            variables: {
                id: fakePostObj.id
            }
        },
        result:  {
            data: {
                Post: {
                    id: fakePostObj.id,
                    content: fakePostObj.content,
                    title: fakePostObj.title,
                    user: {
                        id: fakePostObj.user.id,
                        name: fakePostObj.user.name,
                        username: fakePostObj.user.username,
                    },
                    createdAt: fakePostObj.createdAt,
                    votes: [],
                    post_slug: fakePostObj.post_slug,
                    link: fakePostObj.link,
                    removed: fakePostObj.removed,
                    subreddit: {
                        slug: fakePostObj.subreddit.slug
                    },
                    type: fakePostObj.type,
                    comments: []
                }
            }
        }
    }
  ];

describe('It switches subreddit content based on props', () => {
    it('renders an error with no props', () => {
        const { container, debug } = render(
            <MockedProvider>
              <SubredditContent />
            </MockedProvider>
          );

          expect(container).toMatchSnapshot();
    });

    it('renders an error with the wrong props', async () => {
        const { container, debug } = render(
            <MockedProvider>
              <SubredditContent type="Comment"/>
            </MockedProvider>
          );

          await act(async() => {
            await new Promise(resolve => setTimeout(resolve, 0));
        })

          expect(container).toMatchSnapshot();
    });

    it ('renders the home (through subreddit page)', async () => {
        
        const { container, debug } = render(
            <MockedProvider mocks={mocks} >
                <SubredditPage type="Home" slug={fakeSubredditObj.slug} />
            </MockedProvider>
        );

        await act(async() => {
            await new Promise(resolve => setTimeout(resolve, 0));
        })

        expect(container).toMatchSnapshot();
    });

    it ('renders a specific post', async () => {
        const { container, debug } = render(
            <MockedProvider mocks={mocks} >
                <SubredditContent type="comment" slug={fakePostObj.subreddit.slug} postslug={fakePostObj.post_slug} postid={fakePostObj.id} subreddit={fakePostObj.subreddit}/>
            </MockedProvider>
        );
        await act(async() => {
            await new Promise(resolve => setTimeout(resolve, 0));
        })

        expect(container).toMatchSnapshot();
    });

    it ('renders a perma-linked comment', async () => {
        const { container, debug} = render(
            <MockedProvider mocks={mocks} >
                <SubredditContent type="permalinkedcomments" slug={fakePostObj.subreddit.slug} postslug={fakePostObj.post_slug} postid={fakePostObj.id} subreddit={fakePostObj.subreddit}/>
            </MockedProvider>
        
        
        )

        await act(async() => {
            await new Promise(resolve => setTimeout(resolve, 0));
        })

        expect(container).toMatchSnapshot();
    });

    it ('tries to renders a form with self text with being logged in', async () => {
        const { container, debug} = render(
            <MockedProvider mocks={signedInMocks} >
                <SubredditContent type="form" selftext="true" id={fakePostObj.subreddit.id} slug={fakePostObj.subreddit.slug} postslug={fakePostObj.post_slug} postid={fakePostObj.id} subreddit={fakePostObj.subreddit}/>
            </MockedProvider>
         )

        await act(async() => {
            await new Promise(resolve => setTimeout(resolve, 0));
        })

        expect(container).toMatchSnapshot();
    })

    it ('tries to renders a form with self text without being logged in', async () => {

        const { container, debug} = render(
            <MockedProvider mocks={mocks} >
                <SubredditContent type="form" selftext="true" id={fakePostObj.subreddit.id} slug={fakePostObj.subreddit.slug} postslug={fakePostObj.post_slug} postid={fakePostObj.id} subreddit={fakePostObj.subreddit}/>
            </MockedProvider>
         )

        await act(async() => {
            await new Promise(resolve => setTimeout(resolve, 0));
        });

        expect(container).toMatchSnapshot();
    })

    it ('tries to renders a different form without self text without being logged in', async () => {
        const { container, debug} = render(
            <MockedProvider mocks={mocks} >
                <SubredditContent type="form" id={fakePostObj.subreddit.id} slug={fakePostObj.subreddit.slug} postslug={fakePostObj.post_slug} postid={fakePostObj.id} subreddit={fakePostObj.subreddit}/>
            </MockedProvider>
         )

        await act(async() => {
            await new Promise(resolve => setTimeout(resolve, 0));
        })

        expect(container).toMatchSnapshot();
    })

    it ('tries to renders a different form without self text with being logged in', async () => {
        const { container, debug} = render(
            <MockedProvider mocks={signedInMocks} >
                <SubredditContent type="form" id={fakePostObj.subreddit.id} slug={fakePostObj.subreddit.slug} postslug={fakePostObj.post_slug} postid={fakePostObj.id} subreddit={fakePostObj.subreddit}/>
            </MockedProvider>
         )

         const { container: container2, debug: debug2} = render(
            <MockedProvider mocks={signedInMocks} >
                <SubredditContent type="form" selftext="true" id={fakePostObj.subreddit.id} slug={fakePostObj.subreddit.slug} postslug={fakePostObj.post_slug} postid={fakePostObj.id} subreddit={fakePostObj.subreddit}/>
            </MockedProvider>
         )

        await act(async() => {
            await new Promise(resolve => setTimeout(resolve, 0));
        })

        expect(container).toMatchSnapshot();
        expect(container).not.toEqual(container2);
    })
})