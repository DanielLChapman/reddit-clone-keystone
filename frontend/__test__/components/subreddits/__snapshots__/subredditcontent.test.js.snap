// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It switches subreddit content based on props renders a perma-linked comment 1`] = `
<div>
  <span>
    Something was missing here!
  </span>
</div>
`;

exports[`It switches subreddit content based on props renders a specific post 1`] = `
<div>
  <div
    class="comment-page"
  >
    <div
      class="comment-top-bar"
    >
      <section
        class="comment-arrows"
      >
        <svg
          class="upvote-arrow "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z"
          />
        </svg>
        <span>
          0
        </span>
        <svg
          class="downvote-arrow "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M168 345.941V44c0-6.627-5.373-12-12-12h-56c-6.627 0-12 5.373-12 12v301.941H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.569 9.373 33.941 0l86.059-86.059c15.119-15.119 4.411-40.971-16.971-40.971H168z"
          />
        </svg>
      </section>
      <section
        class="post-title"
      >
        Possimus earum animi
        ...
      </section>
      <section
        class="post-close"
      >
        <a
          href="/r/eius dolorem amet quia quia aut sint"
        >
          × Close
        </a>
      </section>
    </div>
    <section
      class="comment-container"
    >
      <section
        class="comment-page-container"
      >
        <section
          class="comment-page-left"
        >
          <section
            class="reddit-post"
          >
            <section
              class="reddit-post-left"
            >
              <svg
                class="upvote-arrow "
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 256 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z"
                />
              </svg>
              <div
                aria-disabled="false"
                class="upvote-arrow-box"
              />
              <br />
              <span>
                0
              </span>
              <br />
              <svg
                class="upvote-arrow "
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 256 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M168 345.941V44c0-6.627-5.373-12-12-12h-56c-6.627 0-12 5.373-12 12v301.941H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.569 9.373 33.941 0l86.059-86.059c15.119-15.119 4.411-40.971-16.971-40.971H168z"
                />
              </svg>
              <div
                aria-disabled="false"
                class="downvote-arrow-box"
              />
            </section>
            <section
              class="reddit-post-right"
            >
              <section
                class="reddit-post-right-top"
              >
                <span>
                  Posted by 
                  <a
                    class="subreddit-link"
                    href="/user/Mrs. Palma Marks"
                  >
                    u/
                    Mrs. Palma Marks
                  </a>
                </span>
                 
                <span>
                  Posted 
                  0 minutes ago
                </span>
              </section>
              <section
                class="reddit-post-right-bottom"
              >
                <h6>
                  Possimus earum animi quis expedita et optio.
                </h6>
                <div
                  style="display: block;"
                >
                  <p>
                    Velit voluptate necessitatibus excepturi et qui sunt nesciunt provident.
                  </p>
                </div>
              </section>
              <section
                class="reddit-post-right-bottom-footer"
              >
                <a
                  class="subreddit-link"
                  href="/r/eius dolorem amet quia quia aut sint/comments/95b4a3df-d831-4a85-a821-032af032630d/aliquid ea animi architecto accusantium vel enim"
                >
                  0
                   Comments
                </a>
                
              </section>
            </section>
          </section>
          <section
            class="comments-container"
          />
          <div>
            Nothing yet...
          </div>
        </section>
        <section
          class="comment-page-right"
        />
      </section>
    </section>
  </div>
</div>
`;

exports[`It switches subreddit content based on props renders an error with no props 1`] = `
<div>
  <span>
    Something went wrong...
  </span>
</div>
`;

exports[`It switches subreddit content based on props renders an error with the wrong props 1`] = `
<div>
  <span>
    Something went wrong...
  </span>
</div>
`;

exports[`It switches subreddit content based on props renders the home (through subreddit page) 1`] = `
<div>
  <section
    class="subreddit-top-bar"
  >
    <section
      class="subreddit-top-bar-content"
    >
      <section
        class="subreddit-top-bar-name"
      >
        <h1>
          modioccaecatiillumlaborumquoseumvoluptate
        </h1>
        <button>
          Join
        </button>
      </section>
      <span>
        r/
        in fugit in rerum libero odit aut
      </span>
      <span
        class="subreddit-right-mobile-view"
      >
         
        View
         Right Side
      </span>
    </section>
  </section>
  <div
    class="subreddit-content"
    data-testid="subreddit-content"
  >
    <span>
      Nothing to see here!
    </span>
  </div>
</div>
`;

exports[`It switches subreddit content based on props tries to renders a different form without self text with being logged in 1`] = `
<div>
  <div>
    <form
      class="SubredditSubmitForm__Form-sc-4m25ol-0 dSAjAw"
    >
      <fieldset>
        <label
          for="title"
        >
          Title 
          <br />
          <input
            id="title"
            name="title"
            required=""
            type="text"
            value=""
          />
        </label>
        <label
          for="link"
        >
          Link 
          <br />
          <input
            id="link"
            name="link"
            required=""
            type="text"
            value=""
          />
        </label>
        <button
          type="submit"
        >
          Create Link/Media Post
        </button>
      </fieldset>
    </form>
  </div>
</div>
`;

exports[`It switches subreddit content based on props tries to renders a different form without self text without being logged in 1`] = `
<div>
  <span>
    You Must Be Logged In
  </span>
</div>
`;

exports[`It switches subreddit content based on props tries to renders a form with self text with being logged in 1`] = `
<div>
  <div>
    <form
      class="SubredditSubmitForm__Form-sc-4m25ol-0 dSAjAw"
    >
      <fieldset>
        <label
          for="title"
        >
          Title 
          <br />
          <input
            id="title"
            name="title"
            required=""
            type="text"
            value=""
          />
        </label>
        <label
          for="content"
        >
          Text: 
          <br />
          <textarea
            id="content"
            name="content"
            placeholder=""
          />
        </label>
        <button
          type="submit"
        >
          Create Text Post
        </button>
      </fieldset>
    </form>
  </div>
</div>
`;

exports[`It switches subreddit content based on props tries to renders a form with self text without being logged in 1`] = `
<div>
  <span>
    You Must Be Logged In
  </span>
</div>
`;
